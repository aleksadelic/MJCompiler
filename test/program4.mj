program MJProgram
	int x; int y;
{

	int k (int a, int b) int c, d; int niz[]; {
		c = 2;
		c++;
		return c + a + b + 2;
	}
	
	void foo(int a, int b) {
		print(a + b);
	}

	void main() {
		print(1);
	
		x = 3;
		x++;
		
		print(x);
		
		x--;
		
		print(x);
		
		y = 4 + 9 + k(1, 2);
		
		k(1, 2);
		
		print(y);
		
		x = 10 - 3;
		
		print(x);
		
		x = -1 + 3;
		
		print(x);
		
		x = 5 * 3 + 4 / 2 + 10 % 3;
		
		print(x);
		
		y = 2;
		
		x = x / 2;
		
		print(x);
		
		print(5 * 2 + 2);
		
		print((2 + 2) * 5);
		
		x = 5;
		
		foo(3, x);
		
		print('a');
	}
}