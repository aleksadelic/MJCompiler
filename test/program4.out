   55    1    4    3   21    7    2   18    9   12   20    8    a    5    5   10    6    4   20   55    1    5    0    5    4    6   10    1    
truefalse   25   44  555    4   70   22  270   44  500    1    2    1    2    4    5    1    2    3    4    5    6    1    2    3    4    5
Completion took 14863 ms